rfcs/RFC-project-rules.md

```diff
@@ -0,0 +1,75 @@
+# Enforcing contribution rules for repository stability with a merging bot
+
+**Authors:**
+
+* Red Hat: @leseb
+
+## Summary
+
+To ensure the stability, security, and maintainability of the project, we propose enforcing the following contribution rules. These guidelines help maintain code quality, prevent accidental regressions, and improve collaboration across contributors.
+
+We recognize that many of these rules may already be considered common sense by experienced contributors. However, making them explicit and automating their enforcement ensures consistency across all contributors, avoids misunderstandings, and streamlines the development process. The goal is not to restrict contributions but to create a smoother workflow that benefits everyone.
+
+Rather than relying solely on manual enforcement, we propose using a merging bot to handle these rules automatically, reducing the risk of human error and keeping the process efficient.
+
+## Proposed Rules & Automated Enforcement
+
+1. Do not push directly to main
+[Branch protection](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/managing-a-branch-protection-rule) settings will prevent direct pushes to main, requiring all changes to go through pull requests (PRs).
+
+2. Do not self-merge
+The merging bot will ensure that a PR has at least one approval from another maintainer before merging.
+
+3. Do not merge if CI is failing
+The bot will block merging if the CI checks do not pass, ensuring that only validated code is integrated. Maintainers with write access can override the bot and merge the PR anyway if needed.
+
+4. Do not use main for development-use a fork instead
```

Comment:
-> main repository? (since `main` is also a branch)

Reply:
Yes, let me rephrase.